### PR TITLE
save and report dalle.pt at the end of each epoch

### PR DESCRIPTION
**Pull request by rom1504**
also add a parameter to allow specifying a different name to avoid overwriting
if running 2 dalle on the same folder

this solves:
* avoid using a lot of disk with one model every 100 steps under wandb folder
* avoid confusion and errors while running 2 models in the same folder

### CODE REVIEW

**Comment by afiaka87** [comment]
Thanks @rom1504 have you tested this yourself?

**Comment by rom1504** [comment]
I did yes, but having someone else run it once too couldn't hurt, if you want to try it!

**Comment by lucidrains** [comment]
@rom1504 nice! do you want to exclude the `.pt` from the filename? (otherwise looks good!)

File: train_dalle.py [hunk 1]
```diff
@@ -435,15 +438,16 @@ def save_model(path):
     if distr_backend.is_root_worker():
         # save trained model to wandb as an artifact every epoch's end
 
+        save_model(DALLE_OUTPUT_FILE_NAME)
```
Comment (janEbert) [inline train_dalle.py:441]: Please move this out of the `if` block so `save_model` is called on every worker.

Comment (rom1504) [reply train_dalle.py:441]: why do we want to save the model in every worker ?
for example that wouldn't work in my case where the file system is shared between workers

**Comment by rom1504** [comment]
@lucidrains changed!

**Comment by rom1504** [comment]
Hi. Should I change more things here ?

File: train_dalle.py (hunk 1 above)
Comment (janEbert) [reply train_dalle.py:441]: Please take a look at the definition of `save_model`. The method has another guard inside. It's important to call this on all workers so DeepSpeed checkpoints are correctly handled.

Comment (rom1504) [reply train_dalle.py]: got it, I made the change

**Comment by rom1504** [comment]
sorry for the delay for making the last changes
I think this is now ready for review/merging

**Comment by afiaka87** [comment]
@lucidrains you have a look at this yet

**Review by afiaka87 — APPROVED** [review]

**Pull request merged**